opentaskspal/src/main/java/org/dmfs/opentaskspal/tasklists/NameData.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.opentaskspal.tasklists;
+
+import android.content.ContentProviderOperation;
+import android.support.annotation.NonNull;
+
+import org.dmfs.android.contentpal.RowData;
+import org.dmfs.android.contentpal.TransactionContext;
+import org.dmfs.tasks.contract.TaskContract;
+
+
+/**
+ * {@link RowData} to set the name of a task list.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class NameData implements RowData<TaskContract.TaskLists>
+{
+    private final CharSequence mName;
+
+
+    public NameData(@NonNull CharSequence name)
+    {
+        mName = name;
+    }
+
+
+    @NonNull
+    @Override
+    public ContentProviderOperation.Builder updatedBuilder(@NonNull TransactionContext transactionContext, @NonNull ContentProviderOperation.Builder builder)
+    {
+        return builder.withValue(TaskContract.TaskLists.LIST_NAME, mName.toString());
```

Comment:
Why not delegate to a `CharSequenceRowData` here?

Reply:
`CharSequenceRowData` allows `null`, so `NameData` wouldn't fail with that. Maybe we need another class for that. ?

Reply:
Hmm.. I am not sure of the correct approach here actually. We don't validate for `null` just use the `@Nullable` `@NonNull` annotations. Should that be enough in these cases? I mean if there wasn't a `.toString()` call here, it wouldn't fail either without a validation.
Similar question come up with `OriginalInstanceSyncIdData` below.

Reply:
apparently we need another delegate which doesn't allow `null`. Or make sure `CharSequenceRowData` doesn't allow `null` and find another, better solution to "erase" values in the database.

Reply:
This comes back to the topic of whether to delete or leave a value alone when the incoming update doesn't have it (which may be represented as `null`). Because that's what makes using `Optional` instead of the `null`s ambiguous as you mentioned before.
I don't know the actual usage in syncadapters..   can the operations maybe be distinguished based on whether they are in 'override' mode or 'fill in the missing data' mode?
If so, could this maybe be a flag? So we could either have an extended `Optional` model with that flag or pass it in the constructor as separate parameter.

Reply:
Updated the class with a TODO about this nullability.